### PR TITLE
Hocs 2615 add audit

### PR DESCRIPTION
**Pull request by philbarton**
Add the standard audit capability.

- Add and AuditClient
- Write to the audit SNS topic
- Update properties

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/queue/ukvi/UKVIComplaintValidator.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        schema = schemaFactory.getSchema(file.toURI());
+        this.objectMapper = objectMapper;
+    }
+
+    public void validate(String jsonBody, String messageId) throws Exception {
+        clientContext.setContext(user, group, messageId);
+        JsonNode json = objectMapper.readTree(jsonBody);
+        Set<ValidationMessage> validationMessages = schema.validate(json);
+        if (validationMessages.isEmpty()) {
+            auditClient.audit(ukviTypeData.getSuccessfulValidationEvent(), null, null, jsonBody);
```
Comment (MarkWrightBjss) [inline src/main/java/uk/gov/digital/ho/hocs/queue/ukvi/UKVIComplaintValidator.java:56]: "null, null" here will leave "caseType" blank, so these won't show in any audit extracts. Are there any MI requirements around this?

**Review by MarkWrightBjss — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/digital/ho/hocs/client/audit/dto/EventType.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.digital.ho.hocs.client.audit.dto;
+
+public enum EventType {
+    UKVI_PAYLOAD_PASSED_VALIDATED,
+    UKVI_PAYLOAD_FAILED_VALIDATED,
+    UKVI_COMPLAINT_CREATED,
```
[the remark this answers is not in the document]
Comment (philbarton) [reply src/main/java/uk/gov/digital/ho/hocs/client/audit/dto/EventType.java:6]: Yeh, happy to change. As you say once you have the caseId it stays in line.
I'll leave the fail validation one.
For the pass validation I'll add the payload to the case_created event. (Successful validation is implicit then)

File: src/main/java/uk/gov/digital/ho/hocs/queue/ukvi/UKVIComplaintValidator.java (hunk 1 above)
Comment (philbarton) [reply src/main/java/uk/gov/digital/ho/hocs/queue/ukvi/UKVIComplaintValidator.java:56]: I'll remove this from here, and add the payload to the caae_created event.
Does that sound ok?

Comment (philbarton) [reply src/main/java/uk/gov/digital/ho/hocs/queue/ukvi/UKVIComplaintValidator.java:56]: I made up the requirements!

File: src/main/java/uk/gov/digital/ho/hocs/client/audit/dto/EventType.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (philbarton) [reply src/main/java/uk/gov/digital/ho/hocs/client/audit/dto/EventType.java:6]: decided to add the suffix : CREATOR, I think it will avoid confusion

**Review by MarkWrightBjss — COMMENTED** [review]
Thanks, looks good.

File: src/main/java/uk/gov/digital/ho/hocs/queue/ukvi/UKVIComplaintValidator.java (hunk 1 above)
Comment (MarkWrightBjss) [reply src/main/java/uk/gov/digital/ho/hocs/queue/ukvi/UKVIComplaintValidator.java]: :-)

**Review by MarkWrightBjss — APPROVED** [review]

**Pull request merged**